src/server/project.ts

```diff
@@ -1622,6 +1622,7 @@ namespace ts.server {
 
         /*@internal*/
         getPackageJsonsVisibleToFile(fileName: string, rootDir?: string): readonly PackageJsonInfo[] {
+            if (this.projectService.serverMode !== LanguageServiceMode.Semantic) return emptyArray;

```

Comment:
Slightly afraid of all the places that are going to need to be patched up in the future to check like this, but I guess that's programming